realm/realm-library/src/main/java/io/realm/OrderedCollectionChange.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+/**
+ * This interface describes the changes made to a collection during the last update.
+ * <p>
+ * {@link OrderedCollectionChange} is passed to the {@link OrderedRealmCollectionChangeListener} which is registered
+ * by {@link RealmResults#addChangeListener(OrderedRealmCollectionChangeListener)}.
+ * <p>
+ * The change information is available in two formats: a simple array of row indices in the collection for each type of
+ * change, and an array of {@link Range}s.
+ */
+public interface OrderedCollectionChange {
+    /**
+     * The indices of objects in the previous version of the collection which have been removed from this one.
+     *
+     * @return the indices array.
+     */
+    long[] getDeletions();
+
+    /**
+     * The indices in the new version of the collection which were newly inserted.
+     *
+     * @return the indices array.
+     */
+    long[] getInsertions();
+
+    /**
+     * The indices in the new version of the collection which were modified.
+     * <p>
+     * For {@link RealmResults}, this means that one or more of the properties of the object at that index were
+     * modified (or an object linked to by that object was modified).
+     *
+     * @return the indices array.
+     */
+    long[] getChanges();
+
+    /**
+     * The ranges of objects in the previous version of the collection which have been removed from this one.
+     *
+     * @return the {@link Range} array.
+     */
+    Range[] getDeletionRanges();
+
+    /**
+     * The ranges of objects in the new version of the collection which were newly inserted.
+     *
+     * @return the {@link Range} array.
+     */
+    Range[] getInsertionRanges();
+
+    /**
+     * The ranges of objects in the new version of the collection which were modified.
+     *
+     * @return the {@link Range} array.
+     */
+    Range[] getChangeRanges();
```

Comment:
Let's not forget that `RecyclerView.Adapter` calls it "changes" too

> final void	notifyDataSetChanged()

> Notify any registered observers that the data set has changed.

---

> final void	notifyItemChanged(int position, Object payload)

Notify any registered observers that the item at position has changed with an optional payload object.

----

> final void	notifyItemChanged(int position)

Notify any registered observers that the item at position has changed.

----

> final void	notifyItemInserted(int position)

Notify any registered observers that the item reflected at position has been newly inserted.

----

> final void	notifyItemMoved(int fromPosition, int toPosition)

Notify any registered observers that the item reflected at fromPosition has been moved to toPosition.

----

> final void	notifyItemRangeChanged(int positionStart, int itemCount, Object payload)

Notify any registered observers that the itemCount items starting at position positionStart have changed.
----

> final void	notifyItemRangeChanged(int positionStart, int itemCount)

Notify any registered observers that the itemCount items starting at position positionStart have changed.

----

> final void	notifyItemRangeInserted(int positionStart, int itemCount)

Notify any registered observers that the currently reflected itemCount items starting at positionStart have been newly inserted.

----

> final void	notifyItemRangeRemoved(int positionStart, int itemCount)

Notify any registered observers that the itemCount items previously located at positionStart have been removed from the data set.

----

> final void	notifyItemRemoved(int position)

Notify any registered observers that the item previously located at position has been removed from the data set.